ssl/ssl_lib.c

```diff
@@ -6677,18 +6677,24 @@ int SSL_client_hello_get_extension_order(SSL *s, uint16_t *exts, size_t *num_ext
 int SSL_client_hello_get0_ext(SSL *s, unsigned int type, const unsigned char **out,
                        size_t *outlen)
 {
-    size_t i;
-    RAW_EXTENSION *r;
+    PACKET pkt, ext_data;
+    unsigned int ext_type;
+
     const SSL_CONNECTION *sc = SSL_CONNECTION_FROM_SSL(s);
 
     if (sc == NULL)
         return 0;
 
     if (sc->clienthello == NULL)
         return 0;
-    for (i = 0; i < sc->clienthello->pre_proc_exts_len; ++i) {
-        r = sc->clienthello->pre_proc_exts + i;
-        if (r->present && r->type == type) {
+
+    pkt = sc->clienthello->extensions;
+    while (PACKET_remaining(&pkt) > 0) {
+        if (!PACKET_get_net_2(&pkt, &ext_type) ||
+            !PACKET_get_length_prefixed_2(&pkt, &ext_data))
+		return 0;
+
+        if (ext_type == type) {
             if (out != NULL)
                 *out = PACKET_data(&r->data);
```

Comment:
This results in a compilation error because you removed the `r` variable

Reply:
fixed